cmd/tools/test/find_altered_tests.go

```diff
@@ -0,0 +1,199 @@
+//
+// The program outputs environment variable assignments in the format `key=value`,
+// where each key corresponds to a modified test suite category, and the value is a
+// pipe-separated list of altered test suites.
+//
+// Usage:
+//   go run find_altered_tests.go -c <category1> -c <category2> ... -s <sourceRef> -t <targetRef>
+
+// CategoryDirs maps test categories to their corresponding directories
+// If you update these, please also update the corresponding dirs in the Makefile
+// FUNCTIONAL_TEST_ROOT
+// FUNCTIONAL_TEST_XDC_ROOT
+// FUNCTIONAL_TEST_NDC_ROOT
+// INTEGRATION_TEST_DIRS
+// UNIT_TEST_DIRS
+var CategoryDirs = map[string][]string{
+	"unit":           {"./client", "./common", "./internal", "./service", "./temporal", "./tools", "./cmd"},
+	"integration":    {"./common/persistence/tests", "./tools/tests", "./temporaltest", "./internal/temporalite"},
+	"functional":     {"./tests"},
+	"functional_ndc": {"./tests/ndc"},
+	"functional_xdc": {"./tests/xdc"},
+}
+
+func main() {
+	var categories multiFlag
+	var sourceRef string
+	var targetRef string
+
+	flag.Var(&categories, "c", "Category of altered tests to find (can specify multiple)")
+	flag.StringVar(&sourceRef, "s", "", "Source Git reference (commit SHA, branch, etc.)")
+	flag.StringVar(&targetRef, "t", "", "Target Git reference (commit SHA, branch, etc.)")
+	flag.Parse()
+
+	if len(categories) == 0 || sourceRef == "" || targetRef == "" {
+		log.Fatalf("Usage: find_altered_tests -c <category1> -c <category2> ... -s <sourceRef> -t <targetRef>")
+	}
+
+	for _, category := range categories {
+		if _, exists := CategoryDirs[category]; !exists {
+			log.Fatalf("Unknown category: %s", category)
+		}
+	}
+
+	modifiedFiles, err := getModifiedTestFiles(sourceRef, targetRef)
+	if err != nil {
+		log.Fatalf("Error getting modified test files: %v", err)
+	}
+
+	// Initialize a map to hold modified suites per category
+	modifiedSuites := make(map[string][]string)
+
+	for _, category := range categories {
+		dirs := CategoryDirs[category]
+		suites, err := findAlteredTestSuites(modifiedFiles, dirs)
+		if err != nil {
+			log.Fatalf("Error finding altered test suites for category %s: %v", category, err)
+		}
+		modifiedSuites[category] = suites
+	}
+
+	for _, category := range categories {
+		dirs := CategoryDirs[category]
+		suites, err := findAlteredTestSuites(modifiedFiles, dirs)
+		if err != nil {
+			log.Fatalf("Error finding altered test suites for category %s: %v", category, err)
+		}
+
+		// Directly join suites and output the result without using a map
+		joinedSuites := strings.Join(suites, "|")
+		fmt.Printf("modified_%s_test_suites=%s\n", category, joinedSuites)
+	}
+}
+
+// multiFlag allows multiple instances of a flag
+type multiFlag []string
+
+func (m *multiFlag) String() string {
+	return strings.Join(*m, ",")
+}
+
+func (m *multiFlag) Set(value string) error {
+	*m = append(*m, value)
+	return nil
+}
```

Comment:
you are not using these. still want them?

Reply:
I am allowing multiple instance of the `-c` flag